File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.cpp
class Solution {
public:
    vector<int> productExceptSelf(vector<int>& nums) {
        int prod = 1;
        int n = nums.size();
        vector<int> result;
        for(int i = 0; i < n; i++) {
            if(i != 0) {
                int temp = nums[0];
                nums[0] = nums[i];
                nums[i] = temp;
            }
            for(int j = 1; j < n; j++){
                prod *= nums[j];
            }
            result.push_back(prod);
            prod=1;
        }
        return result;
    }
};

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        n = len(nums)
        l,r = 0, n-1
        res = []
        lp = [1]*n
        rp = [1]*n

        while l < n:
            if l != 0:
                lp[l] = lp[l-1] * nums[l-1]
            l += 1
            
        while r >= 0:
            if r != n-1:
                rp[r] = rp[r+1] * nums[r+1]
            r -= 1
        
        for i in range(n):
            res.append(lp[i]*rp[i])
        
        return res
        